File: src/main/java/com/finassistmini/service/UserService.java
package com.finassistmini.service;

import com.finassistmini.dto.*;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
@RequiredArgsConstructor
public class UserService {

    private final KeycloakAdminService keycloakAdminService;

    public List<UserResponse> getAllUsers() {
        return keycloakAdminService.findAll();
    }

    public UserResponse getUserById(String id) {
        return keycloakAdminService.findById(id);
    }

    public UserResponse createUser(CreateUserRequest req) {
        return keycloakAdminService.create(req);
    }

    public UserResponse updateUser(String id, UpdateUserRequest req) {
        return keycloakAdminService.update(id, req);
    }

    public void deleteUser(String id) {
        keycloakAdminService.delete(id);
    }

    public void resetPassword(String id, ResetPasswordRequest req) {
        keycloakAdminService.resetPassword(id, req);
    }

    public void updateStatus(String id, boolean enabled) {
        keycloakAdminService.updateStatus(id, enabled);
    }

    public List<String> getRoles(String id) {
        return keycloakAdminService.getRoles(id);
    }

    public void assignRole(String id, String roleName) {
        keycloakAdminService.assignRole(id, roleName);
    }

    public void removeRole(String id, String roleName) {
        keycloakAdminService.removeRole(id, roleName);
    }
}

File: finassist-frontend/src/services/user.service.ts
import api from './api'
import type { AppUser, CreateUserPayload, UpdateUserPayload, ResetPasswordPayload } from '@/types/user'

export const userService = {

    async getUsers(): Promise<AppUser[]> {
        const { data } = await api.get<AppUser[]>('/users')
        return data
    },

    async getUser(id: string): Promise<AppUser> {
        const { data } = await api.get<AppUser>(`/users/${id}`)
        return data
    },

    async createUser(payload: CreateUserPayload): Promise<AppUser> {
        const { data } = await api.post<AppUser>('/users', payload)
        return data
    },

    async updateUser(id: string, payload: UpdateUserPayload): Promise<AppUser> {
        const { data } = await api.put<AppUser>(`/users/${id}`, payload)
        return data
    },

    async deleteUser(id: string): Promise<void> {
        await api.delete(`/users/${id}`)
    },

    async resetPassword(id: string, payload: ResetPasswordPayload): Promise<void> {
        await api.put(`/users/${id}/password`, payload)
    },

    async updateStatus(id: string, enabled: boolean): Promise<void> {
        await api.put(`/users/${id}/status`, { enabled })
    },

    async getRoles(id: string): Promise<string[]> {
        const { data } = await api.get<string[]>(`/users/${id}/roles`)
        return data
    },

    async assignRole(id: string, roleName: string): Promise<void> {
        await api.post(`/users/${id}/roles/${roleName}`)
    },

    async removeRole(id: string, roleName: string): Promise<void> {
        await api.delete(`/users/${id}/roles/${roleName}`)
    },
}